junifer/data/tests/test_masks.py

```diff
@@ -151,3 +171,84 @@ def test_vickery_patil() -> None:
 
     with pytest.raises(ValueError, match=r"find a Vickery-Patil mask "):
         _load_vickery_patil_mask("wrong", resolution=2)
+
+
+def test_get_mask() -> None:
+    """Test the get_mask function."""
+    reader = DefaultDataReader()
+    with OasisVBMTestingDatagrabber() as dg:
+        input = dg["sub-01"]
+        input = reader.fit_transform(input)
+        vbm_gm = input["VBM_GM"]
+        vbm_gm_img = vbm_gm["data"]
+        mask = get_mask(mask="GM_prob0.2", target_data=vbm_gm)
+
+        assert mask.shape == vbm_gm_img.shape
+        assert_array_equal(mask.affine, vbm_gm_img.affine)
+
+        raw_mask_img, _ = load_mask("GM_prob0.2", resolution=1.5)
+        res_mask_img = resample_to_img(
+            raw_mask_img,
+            vbm_gm_img,
+            interpolation="nearest",
+            copy=True,
+        )
+        assert_array_equal(mask.get_fdata(), res_mask_img.get_fdata())
+
+
+def test_mask_callable() -> None:
+    """Test using a callable mask."""
+
+    def ident(x):
+        return x
+
+    _available_masks["identity"] = {"family": "Callable", "func": ident}
+    reader = DefaultDataReader()
+    with OasisVBMTestingDatagrabber() as dg:
+        input = dg["sub-01"]
+        input = reader.fit_transform(input)
+        vbm_gm = input["VBM_GM"]
+        vbm_gm_img = vbm_gm["data"]
+        mask = get_mask(mask="identity", target_data=vbm_gm)
+
+        assert_array_equal(mask.get_fdata(), vbm_gm_img.get_fdata())
+
+    del _available_masks["identity"]
+
+
+def test_nilearn_compute_masks() -> None:
```

Comment:
Is it possible to parametrize this test?

Reply:
If you parametrize the tests, it will essentially run as the structure as you have now but more atomic and cleaner. What do you mean by "computationally inefficient"? For the ones you require extra steps, it might be worth putting it in other functions. IMO, this makes it maintainable and easy for others to approach it.

Reply:
done